### PR TITLE
Centralizes code and creates a more uniform permission system.

### PR DESCRIPTION
**Pull request by kevgliss**

### CODE REVIEW

File: src/dispatch/auth/permissons.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        class TeapotUserAgentPermission(BasePermission):
+
+            def has_required_permissions(self, request: Request) -> bool:
+                return request.headers.get('User-Agent') == "Teapot v1.0"
+
+    """
+
+    error_msg = "Forbidden."
+    status_code = HTTP_403_FORBIDDEN
+    error_code = HTTP_403_FORBIDDEN
```
Comment (mvilanova) [inline src/dispatch/auth/permissons.py:41]: Do we need it? It's not used anywhere.

File: src/dispatch/auth/permissons.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    .. code-block:: python
+
+        class TeapotUserAgentPermission(BasePermission):
+
+            def has_required_permissions(self, request: Request) -> bool:
+                return request.headers.get('User-Agent') == "Teapot v1.0"
+
+    """
+
+    error_msg = "Forbidden."
```
Comment (mvilanova) [inline src/dispatch/auth/permissons.py:39]: Should we customize this error message based on what the user tries to do?

**Review by mvilanova — APPROVED** [review]
Looks good. Just some nits.

File: src/dispatch/auth/permissons.py (hunk 2 above)
Comment (kevgliss) [reply src/dispatch/auth/permissons.py:39]: We can override this when we construct the permission. But this is the default.

**Pull request merged**